GliaWidgets/Sources/ViewController/Survey/Components/SingleChoiceQuestionView/Survey.SingleChoiceQuestionView.swift

```diff
@@ -85,9 +85,34 @@ extension Survey {
             zip(props.options, optionsStack.arrangedSubviews)
                 .forEach { opt, view in
                     guard let checkboxView = view as? CheckboxView else { return }
+
+                    var state: CheckboxView.State
```

Comment:
I think it will be beneficial to extract this logic into pure static function and also cover it with unit tests. Ideally that logic should not be a part of view but view model, but there is no such for Survey, as I understand.

Reply:
i may be wrong, but could this switch be simplified with if statement like this? will it not be less code and more straightforward what happens inside?
```swift
                    // If user selected or it should be selected by default.
                    let isSelected = props.selected == opt || props.defaultOption == opt

                    // Handle first selection and apply default selection
                    if isSelected, props.selected == nil {
                        opt.select(opt)
                    }
```

Reply:
First I added the tests and then I exchanged it for Yurii's code, everything works c3c6c15fa0eac9f067a7a077e86d0692ca449c71